gcloud-java-core/src/main/java/com/google/gcloud/ServiceOptions.java

```diff
@@ -479,6 +487,20 @@ protected static String getAppEngineProjectId() {
     }
   }
 
+  protected static String serviceAccountProjectId() {
+    String project = null;
+    String credentialsPath = System.getenv("GOOGLE_APPLICATION_CREDENTIALS");
+    if(credentialsPath != null) {
+      try (InputStream credentialsStream = new FileInputStream(credentialsPath)) {
```

Comment:
One could set it like [so] (https://cloud.google.com/appengine/docs/java/config/appconfig#Java_appengine_web_xml_System_properties_and_environment_variables). However in this case, that would not be a good idea.